R/delay_distributions.R

```diff
@@ -209,44 +238,98 @@ mcs_delay <- function(
 
   distribution <- match.arg(distribution)
 
-  # Generate integer that sets the seed (if NULL) in set.seed() function.
+  # convert date_1 and date_2 to lists if they are vectors:
+  if (!is.list(date_1)) date_1 <- list(date_1)
+  if (!is.list(date_2)) date_2 <- list(date_2)
+
+  # Step 1: Parameter estimation using complete cases:
+  par_list <- purrr::map2(
+    date_1,
+    date_2,
+    dist_helper,
+    distribution = distribution
+  )
+
+  # Step 2: Simulation of random numbers:
+  ## Generate integer that sets the seed (if NULL) in set.seed() function.
   if (purrr::is_null(seed)) {
     seed <- as.integer(stats::runif(n = 1, min = 0, max = 1e6))
```

Comment:
But what if the user wants to make the simulation result reproducible afterwards, i.e. do MCS and then wants to know the seed of how this result was obtained? 
I would say that this isn't possible, since "set.seed returns NULL, invisibly.". 

But I see your point, with seed null it would be a "cleaner" random number generation